added for the n'th time

ChadiSeif opened the pull request.

ChadiSeif committed:
added for the n'th time

ChadiSeif committed:
Create day3.js

raminos commented:
great it worked 🚀 ! Will take a look at it later 😊

ChadiSeif commented:
Ohhh thanks god !
Appreciate your help Ramin

Le jeu. 20 janv. 2022 à 13:12, Ramin Azhdari ***@***.***> a
écrit :

> great it worked 🚀 ! Will take a look at it later 😊
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ChadiSeif/adventofcode/pull/1#issuecomment-1017434549>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUCJZ5IHGII2ENPIOFOKRY3UW73ZXANCNFSM5MMSB5QA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

raminos commented:
first things first. You should run node from the command line or if you have difficulties with using the shell on Windows use an online IDE that creates a solid environment for you like https://repl.it

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+
+process.stdin.on('data', function(data) { 
+    
+    var test = data.split("\n")
```
In modern JavaScript we are using `let` and `const` instead of `var`. We only use `var` if really needed.

```suggestion
    const test = data.split("\n")
```

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+		zero=0
+		i++
+	 }
+	 console.log(parseInt(gamma,2)*parseInt(epsilon,2))
+	 
+    });
```
Your function should return a value. Afterwards you can log it.

```suggestion
	 return parseInt(gamma,2) * parseInt(epsilon,2)
    });
```

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+	var gamma= "";
+	var epsilon =""
+		let i = 0
+	 while ( i < test[1].length )  {
+		for(let j=0 ; j<test.length ; j++)	{		
```
Why do you use first `while` and then `for`? This makes me as a reviewer wonder if you want to achieve different things, but you actually aren't.

```suggestion
	 for (let i = 0; i < test[1].length; i++)  {
		for (let j = 0; j < test.length; j++)	{		
```

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+		for(let j=0 ; j<test.length ; j++)	{		
+		if (test[j][i] === "0"){
+			zero += 1
+		} else one += 1 
+		}
```
Definitely learn how to format your code well. This is very hard to read and a reviewer could have missed that the bracket was wrongly indented

```suggestion
		} else {
		  one += 1 
		}
              }
```

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+		if ( one > zero ){
+		gamma = gamma + "1" ;
+		epsilon = epsilon +"0"
+			}else {gamma = gamma +"0";
+			epsilon = epsilon +"1"}
+			;
```
Same as above.

```suggestion
			} else {
			  gamma = gamma + "0";
			  epsilon = epsilon + "1";
			}
```

raminos commented on day3-b.js:
```diff
@@ -0,0 +1,33 @@
+process.stdin.setEncoding('utf8');
+
+
+process.stdin.on('data', function(data) { 
```
Always use the opportunity to name your functions. As a reviewer I cannot easily understand what this function tries to achieve.

```suggestion
process.stdin.on('data', function getPowerConsumption(data) { 
```

raminos reviewed (commented):
It is great that the code works! 🎉  Congrats!

Now you need to make the code readable for others 🙃  this is the **key** skill for a professional developer.

1. Your code should be well formatted. You can use formatters and linters to help you (see prettier and eslint)
2. You should give meaningful names to your variables and functions (eg "gammaRate" instead of "gamma")
3. Try to show intent with you code. Your code is very "imperative" and could have been more "declarative" (see [this video](https://www.youtube.com/watch?v=E7Fbf7R3x6I for an explanation)

Another thing. Functions always return something (or they create side effects). We never use a function to call a `console.log`. You can use platforms like leetcode and exercism to train this habit.

If you want to write testable code you will need to return values. You can try to use `jest` and test functions like this one.

If you want you can improve you code and I give it a second look 😊

ChadiSeif commented:
this is great Ramin, i really appreciate your comments, this is so
beneficial for me!
It is so good to know what to improve, i will focus more focused on these
points, thanks for the effort you have done !

Le jeu. 20 janv. 2022 à 16:32, Ramin Azhdari ***@***.***> a
écrit :

> ***@***.**** commented on this pull request.
>
> It is great that the code works! 🎉 Congrats!
>
> Now you need to make the code readable for others 🙃 this is the *key*
> skill for a professional developer.
>
>    1. Your code should be well formatted. You can use formatters and
>    linters to help you (see prettier and eslint)
>    2. You should give meaningful names to your variables and functions
>    (eg "gammaRate" instead of "gamma")
>    3. Try to show intent with you code. Your code is very "imperative"
>    and could have been more "declarative" (see [this video](
>    https://www.youtube.com/watch?v=E7Fbf7R3x6I for an explanation)
>
> If you want you can improve you code and I give it a second look 😊
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788862399>:
>
> > @@ -0,0 +1,33 @@
>
> +//**Hello Ramin this is part 1 of day 3 :D **//
>
> +//i am running this online to get the input here : https://www.codechef.com/ide //
>
> +process.stdin.resume();
>
> +process.stdin.setEncoding('utf8');
>
> +
>
> +
>
> +process.stdin.on('data', function(data) {
>
> +
>
> +    var test = data.split("\n")
>
>
> In modern JavaScript we are using let and const instead of var. We only
> use var if really needed.
> ⬇️ Suggested change
>
> -    var test = data.split("\n")
>
> +    const test = data.split("\n")
>
>
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788863732>:
>
> > +	 console.log(parseInt(gamma,2)*parseInt(epsilon,2))
>
> +	
>
> +    });
>
>
> Your function should return a value. Afterwards you can log it.
> ⬇️ Suggested change
>
> -	 console.log(parseInt(gamma,2)*parseInt(epsilon,2))
>
> -	
>
> -    });
>
> +	 return parseInt(gamma,2) * parseInt(epsilon,2)
>
> +    });
>
>
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788867382>:
>
> > +	 while ( i < test[1].length )  {
>
> +		for(let j=0 ; j<test.length ; j++)	{		
>
>
> Why do you use first while and then for? This makes me as a reviewer
> wonder if you want to achieve different things, but you actually aren't.
> ⬇️ Suggested change
>
> -	 while ( i < test[1].length )  {
>
> -		for(let j=0 ; j<test.length ; j++)	{		
>
> +	 for (let i = 0; i < test[1].length; i++)  {
>
> +		for (let j = 0; j < test.length; j++)	{		
>
>
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788869500>:
>
> > +		} else one += 1
>
> +		}
>
>
> Definitely learn how to format your code well. This is very hard to read
> and a reviewer could have missed that the bracket was wrongly indented
> ⬇️ Suggested change
>
> -		} else one += 1
>
> -		}
>
> +		} else {
>
> +		  one += 1
>
> +		}
>
> +              }
>
>
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788870887>:
>
> > +			}else {gamma = gamma +"0";
>
> +			epsilon = epsilon +"1"}
>
> +			;
>
>
> Same as above.
> ⬇️ Suggested change
>
> -			}else {gamma = gamma +"0";
>
> -			epsilon = epsilon +"1"}
>
> -			;
>
> +			} else {
>
> +			  gamma = gamma + "0";
>
> +			  epsilon = epsilon + "1";
>
> +			}
>
>
> ------------------------------
>
> In day3-b.js
> <https://github.com/ChadiSeif/adventofcode/pull/1#discussion_r788873666>:
>
> > @@ -0,0 +1,33 @@
>
> +//**Hello Ramin this is part 1 of day 3 :D **//
>
> +//i am running this online to get the input here : https://www.codechef.com/ide //
>
> +process.stdin.resume();
>
> +process.stdin.setEncoding('utf8');
>
> +
>
> +
>
> +process.stdin.on('data', function(data) {
>
>
> Always use the opportunity to name your functions. As a reviewer I cannot
> easily understand what this function tries to achieve.
> ⬇️ Suggested change
>
> -process.stdin.on('data', function(data) {
>
> +process.stdin.on('data', function getPowerConsumption(data) {
>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ChadiSeif/adventofcode/pull/1#pullrequestreview-858389055>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUCJZ5MNWC2UUJMZO2JLCADUXATKDANCNFSM5MMSB5QA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif committed:
Update day3-b.js

Co-authored-by: Ramin Azhdari <47226721+raminos@users.noreply.github.com>

ChadiSeif reviewed (commented):
Thanks for your efforts !

raminos commented:
> Thanks for your efforts !

Excited to see the next one ;)
